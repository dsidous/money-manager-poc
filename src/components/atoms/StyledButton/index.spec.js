import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { StyledButton } from '.';

configure({ adapter: new Adapter() });

describe('Styled Button', () => {
  it('renders as expected', () => {
    const render = shallow(<StyledButton />);
    expect(render).toMatchSnapshot();
  });

  it('Test press event', () => {
    const mockCallBack = jest.fn();

    const render = shallow((<StyledButton onPress={mockCallBack}>TEST</StyledButton>));
    render.find('Styled(TouchableOpacity)').simulate('press');
    expect(mockCallBack).toHaveBeenCalled();
  });

  describe('renders with disabled prop', () => {
    const mockCallBack = jest.fn();
    const render = shallow(<StyledButton disabled onPress={mockCallBack} />);

    it('renders as expected', () => {
      expect(render).toMatchSnapshot();
    });

    it('Disable press event', () => {
      render.find('Styled(TouchableOpacity)').simulate('press');
      expect(mockCallBack).not.toHaveBeenCalled();
    });
  });
});
