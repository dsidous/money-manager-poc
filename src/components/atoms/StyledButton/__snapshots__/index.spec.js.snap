// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Button renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Text) />,
      "disabled": undefined,
      "onPress": undefined,
      "outlined": undefined,
      "style": undefined,
      "underlayColor": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "disabled": undefined,
        "outlined": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text) />,
        "disabled": undefined,
        "onPress": undefined,
        "outlined": undefined,
        "style": undefined,
        "underlayColor": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "disabled": undefined,
          "outlined": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Styled Button renders with disabled prop renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledButton
    disabled={true}
    onPress={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Styled(Text)
        disabled={true}
      />,
      "disabled": true,
      "onPress": null,
      "outlined": undefined,
      "style": undefined,
      "underlayColor": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": undefined,
        "disabled": true,
        "outlined": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Styled(Text)
          disabled={true}
        />,
        "disabled": true,
        "onPress": null,
        "outlined": undefined,
        "style": undefined,
        "underlayColor": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": undefined,
          "disabled": true,
          "outlined": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
